etl/src/main/java/org/pih/hivmigration/etl/sql/Setup.groovy

```diff
@@ -16,16 +16,23 @@ class Setup extends SqlMigrator {
             DELIMITER ;
         ''')
 
-        executeMysql('''
-            -- Checks whether a string value represents a number. Allows trailing whitespace, but not leading whitespace.
-            -- from https://stackoverflow.com/a/5065007/1464495
+        executeMysql("Create function is_number", '''
+            -- Checks whether a string value represents a number. Allows surrounding whitespace.
+            -- Allows an arbitrary number of decimal points or commas.
+            -- Recommended to use with 'extract_number', e.g.
+            --   SELECT extract_number(result) FROM tbl WHERE is_number(result)
             DROP FUNCTION IF EXISTS is_number;
             CREATE FUNCTION is_number (_value text) RETURNS boolean DETERMINISTIC
-            RETURN concat('', _value * 1) = _value;
+                RETURN _value REGEXP '^[0-9\\.,]+$' AND _value REGEXP '[0-9]';
```

Comment:
My REGEXP skills are fading :) ... what does the first match?  It looks like a NOT match  on 0-9 and , and . but I figure I'm wrong about that?

I do think it's clearer to have a REGEX than the weird concat thing...

Reply:
`^` means beginning. `$` means end. `^ only means "not the following" when inside and at the start of [] , so [^...]` (from SO).

So this is "any character 0-9, or '.', or ','"